src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/order/AddOrderCommandParser.java

```diff
@@ -15,17 +18,31 @@
 public class AddOrderCommandParser implements Parser<AddOrderCommand> {
     @Override
     public AddOrderCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CLIENT_PHONE);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CLIENT, PREFIX_CLIENT_PHONE,
+                PREFIX_CLIENT_NAME, PREFIX_CLIENT_ADDRESS);
 
-        if (!ParserUtil.areAllPrefixesPresent(argMultimap, PREFIX_CLIENT_PHONE)
+        if ((!ParserUtil.areAllPrefixesPresent(argMultimap, PREFIX_CLIENT)
+                && !ParserUtil.areAllPrefixesPresent(argMultimap,
+                                                    PREFIX_CLIENT_PHONE, PREFIX_CLIENT_NAME, PREFIX_CLIENT_ADDRESS))
```

Comment:
Wondering if we should remove the all prefixes present check because this will become complicated once we have recipe bookmarks. Could you abstract these out for now so that it is easier to tell what this does?

```suggestion
        boolean isClientIndexPresent = ParserUtil.areAllPrefixesPresent(argMultimap, PREFIX_CLIENT);
        boolean areClientFieldsPresent = ParserUtil.areAllPrefixesPresent(argMultimap,
                PREFIX_CLIENT_PHONE, PREFIX_CLIENT_NAME, PREFIX_CLIENT_ADDRESS);

        if ((!isClientIndexPresent && !areClientFieldsPresent)
```